### ISSUE TITLE
Flying units jitter around the main ship

### ISSUE THREAD

**Issue by FinishedFragment** [Anuken/Mindustry#4190]
**Platform**: Android

**Build**: bleeding-edge 20453

**Issue**: When playing on a map in campaign mode and double-tapping your own ship to make units follow you, the units become "jiggly".

**Steps to reproduce**: See attached save for the **Windswept Islands**. I tried reproducing this on a custom game but for whatever reason I can not get more than 3 Poly units to follow my ship. The effect that is present on Windswept Islands is still there, but very subtle. Move your ship around a bit if the effect does not become visible.
- Produce **four or more** flying units of the same type. I used Horizon and Poly units to test. I am not sure about other units.
- Double-tap your ship to get the 4 units to follow you in a formation
- The effect will be visible. I am not sure how I can accurately describe that, but it is some variant of jittering and twitching, which is quite annoying.

Also, probably an unrelated issue, the spawned enemies in the wave in the save (no rhyme intended) do not move. I can open a separate issue for this if needed.

**Link(s) to mod(s) used**: -

**Save file**: [save.zip](https://github.com/Anuken/Mindustry/files/5756013/save.zip)
I am not sure how I can only include the save for Windswept Islands instead of everything, but I prefer to not have my soul reaped for sharing an incomplete save.

I tested loading the save and it will be necessary to double-tap the ship because the units "detached" from the ship after save and quit.

**(Crash) logs**: not a crash

---
 
- [x] **I have updated to the latest release (https://github.com/Anuken/Mindustry/releases) to make sure my issue has not been fixed.**
- [x] **I have searched the closed and open issues to make sure that this problem has not already been reported.**

**Comment by Anuken** [comment]
Cannot reproduce. Attempted with several different types of units and followers, and the most I could see was single-pixel jitter - nothing that could affect gameplay.

> Also, probably an unrelated issue, the spawned enemies in the wave in the save do not move

Naval pathfinding takes some time to calculate.

**Comment by FinishedFragment** [comment]
I think I figured out why that happens. It seems to be tied to the framerate, which makes it a potential non-issue. When there are areas where I have low fps (15-20 fps), the jittering is much, much stronger.

In order to reproduce this I reproduced a lot of units in a custom game to make the fps drop.

>Naval pathfinding takes some time to calculate.

The naval enemy unit did not move the entire wave.

**Comment by Anuken** [comment]
> The naval enemy unit did not move the entire wave.

If your device is slow enough that the units on the save cause FPS drops, that's not surprising.

**Comment by BlueWolf3682** [comment]
this has been successfully reproduced on my ipod touch 6. units turn back and forth quite fast and settle down only after a while. only happens with faster flying units

**Comment by FinishedFragment** [comment]
>If your device is slow enough that the units on the save cause FPS drops, that's not surprising.

There are indeed some fps drops but only in the areas where my (naval) units accumulate. I get a stable 45-60 fps in any other area of Windswept Islands. The enemy unit moved "properly" earlier but after I added some defenses, and especially some walls they are just stuck there. The enemy naval unit properly attacks my ship if I get near it, but does not move.

**Comment by Anuken** [comment]
It moves for *me*, which implies your device is struggling with the map.
